x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/percentiles_agg/percentiles_form_component.tsx

```diff
@@ -5,38 +5,103 @@
  * 2.0.
  */
 
-import React from 'react';
+import React, { useCallback, useMemo } from 'react';
 import { i18n } from '@kbn/i18n';
-import { EuiFieldText, EuiFormRow } from '@elastic/eui';
-import type { IPivotAggsConfigPercentiles } from './types';
+import type { EuiComboBoxOptionOption } from '@elastic/eui';
+import { EuiComboBox, EuiFormRow } from '@elastic/eui';
+import type { IPivotAggsConfigPercentiles, ValidationResultErrorType } from './types';
+
+const ERROR_MESSAGES: Record<ValidationResultErrorType, string> = {
+  INVALID_FORMAT: i18n.translate('xpack.transform.agg.popoverForm.invalidFormatError', {
+    defaultMessage: 'Percentile must be a valid number',
+  }),
+  PERCENTILE_OUT_OF_RANGE: i18n.translate(
+    'xpack.transform.agg.popoverForm.percentileOutOfRangeError',
+    {
+      defaultMessage: 'Percentiles must be between 0 and 100',
+    }
+  ),
+};
 
 export const PercentilesAggForm: IPivotAggsConfigPercentiles['AggFormComponent'] = ({
   aggConfig,
   onChange,
   isValid,
+  errorMessageType,
 }) => {
+  const selectedOptions = useMemo(
+    () => aggConfig.percents?.map((p) => ({ label: p.toString() })) ?? [],
+    [aggConfig.percents]
+  );
+
+  const handleCreateOption = useCallback(
+    (inputValue: string) => {
+      if (!isValid) return false;
+
+      const newValue = Number(inputValue.replace(',', '.'));
+
+      // Skip if this percentile value already exists after normalizing number formats
+      // e.g. "+12" and "12" are considered the same value
+      if (aggConfig.percents?.includes(newValue)) {
+        return false;
+      }
+
+      const newOption = {
+        label: newValue.toString(),
+      };
+      const updatedOptions = [...selectedOptions, newOption];
+
+      onChange({
+        percents: updatedOptions.map((option) => Number(option.label)),
+      });
+    },
+    [isValid, onChange, selectedOptions, aggConfig.percents]
+  );
+
+  const handleOptionsChange = useCallback(
+    (newOptions: Array<EuiComboBoxOptionOption<string>>) => {
+      onChange({ percents: newOptions.map((option) => Number(option.label)) });
+    },
+    [onChange]
+  );
+
+  const handleSearchChange = useCallback(
+    (searchValue: string) => {
+      // If we're clearing the input after a valid creation,
+      // this is the post-creation cleanup
+      if (searchValue === '' && aggConfig.pendingPercentileInput && isValid) return;
+
+      onChange({
+        ...aggConfig,
+        pendingPercentileInput: searchValue,
+      });
+    },
+    [aggConfig, onChange, isValid]
+  );
+
+  const getErrorMessage = () => {
+    if (!isValid && errorMessageType) {
+      return ERROR_MESSAGES[errorMessageType];
+    }
+  };
```

Comment:
not need making it a funciton 
```suggestion
  const errorMessage = isValid && !errorMessageType ? undefined : ERROR_MESSAGES[errorMessageType];
```

Reply:
Done similarly in: [#a2f0df3](https://github.com/elastic/kibana/pull/197816/commits/a2f0df329ddc2cdb340e002a96ef1cf9573b4713)